api/tokens.go

```diff
@@ -162,6 +170,51 @@ func (api *API) SignOutHandler(ctx context.Context) http.HandlerFunc {
 	}
 }
 
+func (api *API) RefreshHandler(ctx context.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		var errorList []apierrorsdeprecated.Error
+
+		refreshToken := models.RefreshToken{
+			TokenString: req.Header.Get(RefreshTokenHeaderName),
+		}
+		refreshToken.Validate(ctx, &errorList)
```

Comment:
I've updated the Validate methods to return errorlist.

Reply:
Ah I also meant that because you can just return the errors you need you dont need to pass errorList in at all. Hence you'd need to append to the error list in the scope of your handler, because your refreshToken.Validate only returns the errors for itself (and in fact could just return singular instead of [] I think). 

This might come under the error handling work youve mentioned though.